### ISSUE TITLE
Range-Doppler Data Difference Between taken from mmwave-studio and taken from this code

### ISSUE THREAD

**Issue by Khan9944** [AndyYu0010/real-time-radar#18]
Hello,

We are working on an AI-based project using the IWR1843BOOST board. In this project, we previously collected data using mmWave Studio. Below, you can see two GIFs showing Range-Doppler data obtained with mmWave Studio, each containing 128 frames.

The first image (mmwave_studio_no_person.gif) shows Range-Doppler data when no person is in front of the radar.

![Image](https://github.com/user-attachments/assets/c4cbb473-4af0-42dc-a2b2-bbff8e703b72)

The second image (mmwave_studio_walking_front_and_back.gif) shows Range-Doppler data when a person is walking forward and backward in front of the radar.

![Image](https://github.com/user-attachments/assets/21a53aaa-49dd-4913-b447-d287783f07e9)

And this is range-doppler data we get with your code:

![Image](https://github.com/user-attachments/assets/c12437a7-5b85-4d3d-b246-e917c0cb4d0c)

The data collected with mmWave Studio is saved in .bin format. We then process these binary files using the following code to generate the Range-Doppler images shown above:

`

import os
import numpy as np
import matplotlib.pyplot as plt
from mmwave.dataloader import DCA1000
import mmwave.dsp as dsp

bin_file_path = r"C:\Users\prevl\Desktop\raw_radar"
range_doppler_path = r"C:\Users\prevl\Desktop\raw_radar\visuals"

def process_bin_to_images(file_path, output_folder):
    num_tx_antennas = 2
    numChirpsPerFrame = 128
    numFrames = 128
    numRxAntennas = 4
    numADCSamples = 256

    adc_data = np.fromfile(file_path, dtype=np.uint16)
    adc_data = adc_data.reshape(numFrames, -1)
    adc_data = np.apply_along_axis(DCA1000.organize, 1, adc_data, num_chirps=numChirpsPerFrame, num_rx=numRxAntennas, num_samples=numADCSamples)

    os.makedirs(output_folder, exist_ok=True)

    for i in range(numFrames):
        frame = adc_data[i]
        range_fft_data = np.fft.fft(frame, axis=-1)
        assert range_fft_data.shape == (numChirpsPerFrame, numRxAntennas, numADCSamples), "[ERROR] Radar cube is not the correct shape!"
        fft2d_in = dsp.separate_tx(range_fft_data, num_tx_antennas, vx_axis=1, axis=0)
        fft2d_in = np.transpose(fft2d_in, axes=(2, 1, 0))
        fft2d_out = np.fft.fft(fft2d_in)
        aoa_input = fft2d_out

        fft2d_log_abs = 40 * np.log10(np.abs(fft2d_out) + 1e-10)
        det_matrix = np.sum(fft2d_log_abs, axis=1)
        det_matrix_vis = np.fft.fftshift(det_matrix, axes=1)

        plt.figure(figsize=(12, 6))
        plt.imshow(det_matrix_vis / det_matrix_vis.max(), aspect='auto', cmap='jet')
        plt.title(f"Range-Doppler plot {i} - {os.path.basename(file_path)}")
        plt.colorbar(label='Amplitude (dBFS)')
        plt.xlabel('Range-fft')
        plt.ylabel('Doppler-fft')

        plt.savefig(os.path.join(output_folder, f'RangeDoppler_{i}.png'))
        plt.close()

`

However, the Range-Doppler data we obtained using this method looks quite different from the data generated by your application. When we move in front of the radar, we notice significant discrepancies. When performing a forward/backward walking motion in front of the radar sensor, the Range-Doppler heatmap in your application shows changes only in the central region, and these changes occur in an up-and-down direction. However, in the data we obtained using mmWave Studio, we observe changes over a wider area, and instead of vertical shifts, the variations appear to move sideways.

We have investigated possible reasons for this issue. We tried to match the FFT transformations in your code with the ones we implemented, but we couldn't resolve the differences.

Could you help us understand the cause of this discrepancy?

**Comment by walker930** [comment]
The main difference is that the x-axis and y-axis placement corresponds to range and velocity in your first two images, opposite ours. Please see the figure below from the VTC paper, where velocity-range is used. This is likely the primary reason for the directional differences you described. If you want to modify it, you can directly adjust the code in tools/DSP.py to match your implementation.

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/f9f57817-f87f-4e61-a436-136083605c34" />

Here is a simple video [link](https://www.youtube.com/watch?v=Z6rTQDMe6a4) demonstrating the axis placement. This placement is mainly because our application is gesture recognition. Since we place the radar flat, we use this y-axis orientation for visualization. 

Regarding the difference in variation intensity, the parameters you have set (such as num_tx_antennas, numChirpsPerFrame, numRxAntennas, etc.) are clearly different from the radar configuration used in our project (see config/IWR1843_cfg.cfg), since the application scenario is different. Please try replacing it using your configuration and see if it fits the need.

**Comment by Khan9944** [comment]
Thank you so much for answering. I have a question again. I want to install tensorflow in the environment with the libraries in the requirements.txt file, but I couldn't find a version of tensorflow that will not conflicts with the libraries in requirements.txt. If it's possible, can you provide an available tensorflow version please?

**Comment by loeens** [comment]
@Khan9944 as far as I am concerned you can just update all versions and you don't need to stick with the versions in requirements.txt. I created a fork for the IWRL6432 and didn't notice any issues when updating the versions.
